src/modules/universal-discovery/components/content-table/content.table.component.js

```diff
@@ -0,0 +1,184 @@
+        };
+
+        this.setActivePage = this.setActivePage.bind(this);
+        this.renderItem = this.renderItem.bind(this);
+    }
+
+    componentWillReceiveProps({ items, perPage }) {
+        this.setState(state => ({
+            ...state,
+            items,
+            pages: this.splitToPages(items, perPage)
+        }));
+    }
+
+    /**
+     * Splits items into pages
+     *
+     * @method splitToPages
+     * @param {Array} items
+     * @param {Number} perPage
+     * @returns {Array}
+     * @memberof ContentTableComponent
+     */
+    splitToPages(items, perPage) {
+        return items.reduce((pages, item, index) => {
+            const pageIndex = Math.floor(index / perPage);
+
+            if (!pages[pageIndex]) {
+                pages[pageIndex] = [];
+            }
+
+            pages[pageIndex].push(item);
+
+            return pages;
+        }, []);
+    }
+
+    /**
+     * Sets active page index state
+     *
+     * @method setActivePage
+     * @param {number} activePage
+     * @memberof ContentTableComponent
+     */
+    setActivePage(activePage) {
+        this.setState(state => ({
+            ...state,
+            activePage
+        }));
+    }
+
+    /**
+     * Renders single item
+     *
+     * @method renderItem
+     * @param {Object} item
+     * @returns {Element}
+     * @memberof ContentTableComponent
+     */
+    renderItem(item) {
+        item = item.Location;
+
+        const { contentTypesMap, onItemSelect, onItemClick, labels } = this.props;
+
+        return <ContentTableItemComponent
+            key={item.id}
+            data={item}
+            contentTypesMap={contentTypesMap}
+            onPreview={onItemSelect}
+            onItemClick={onItemClick}
+            labels={labels.item} />;
+    }
+
+    /**
+     * Renders single search results item
+     *
+     * @method renderHeader
+     * @returns {Element}
+     * @memberof ContentTableComponent
+     */
+    renderHeader() {
+        const { labels } = this.props;
+
+        return <ContentTableHeaderComponent
+            labels={labels.header}
+        />;
+    }
+
+    /**
+     * Renders pagination
+     *
+     * @method renderPagination
+     * @returns {Element}
+     * @memberof ContentTableComponent
+     */
+    renderPagination() {
+        const { pages, activePage } = this.state;
+        const { labels } = this.props;
+        const pagesCount = pages.length;
+
+        const paginationAttrs = {
+            minIndex: 0,
+            maxIndex: pagesCount - 1,
+            activeIndex: activePage,
+            onChange: this.setActivePage,
+            labels: labels.pagination
+        };
+
+        if (pagesCount === 0 || paginationAttrs.minIndex === paginationAttrs.maxIndex) {
+            return null;
+        }
+
+        return <ContentTablePaginationComponent {...paginationAttrs} />;
+    }
+
+    render() {
+        const { showWhenNoItems } = this.props;
```

Comment:
Do we want to show empty table? What is the purpose of this?

Reply:
When you have no bookmarks, I think that empty table should be displayed for a user so that he would know that there are no bookmarks. The blank tab would suggest that something is not working or something is loading.

Reply:
So maybe it would be good to display some text "Add content to bookmarks" not the empty table? Probably something to discuss with PMs.

Reply:
Thanks @dew326 for spotting this.
In order to be consistent with the other interfaces of the application, you can display:
`No content items. Content items you bookmark will appear here.`

Reply:
@SylvainGuittard This message should be displayed in empty table, like in Sub-items table, or should it be displayed without table?

Reply:
@tischsoic 
Without the table and using the same style as the meta preview
![image](https://user-images.githubusercontent.com/1157233/40841061-1e7804e4-6577-11e8-8610-dc2706317780.png)
Thanks